eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -0,0 +1,198 @@
+package com.azure.tools.bomgenerator;
+
+import com.azure.tools.bomgenerator.models.BomDependency;
+import com.azure.tools.bomgenerator.models.BomDependencyComparator;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.DependencyManagement;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+import static com.azure.tools.bomgenerator.Utils.AZURE_CORE_GROUPID;
+import static com.azure.tools.bomgenerator.Utils.AZURE_CORE_TEST_LIBRARY;
+import static com.azure.tools.bomgenerator.Utils.AZURE_PERF_LIBRARY_IDENTIFIER;
+import static com.azure.tools.bomgenerator.Utils.AZURE_TEST_LIBRARY_IDENTIFIER;
+import static com.azure.tools.bomgenerator.Utils.EXCLUSION_LIST;
+import static com.azure.tools.bomgenerator.Utils.POM_TYPE;
+import static com.azure.tools.bomgenerator.Utils.SDK_DEPENDENCY_PATTERN;
+import static com.azure.tools.bomgenerator.Utils.SDK_NON_GA_PATTERN;
+
+public class BomGenerator {
+    private String outputFileName;
+    private String inputFileName;
+    private String pomFileName;
+    private String externalDependenciesFileName;
+
+    private static Logger logger = LoggerFactory.getLogger(BomGenerator.class);
+
+    BomGenerator() {
+    }
+
+    public void setInputFile(String inputFileName) {
+        this.inputFileName = inputFileName;
+    }
+
+    public void setOutputFile(String outputFileName) {
+        this.outputFileName = outputFileName;
+    }
+
+    public void setPomFile(String pomFileName) {
+        this.pomFileName = pomFileName;
+    }
+
+    public void setExternalDependenciesFile(String externalDependenciesFileName) {
+        this.externalDependenciesFileName = externalDependenciesFileName;
+    }
+
+
+    public void generate() {
+        TreeSet<BomDependency> inputDependencies = scan();
+        TreeSet<BomDependency> externalDependencies = resolveExternalDependencies();
```

Comment:
Is `TreeSet` being used here to maintain the set in a certain sort order?

Reply:
Mostly because AFAIK treeset is the only supported Java DS that allows comparator in the set.